### DISCUSSION TITLE
Interested in Contributing to JEG Upgrade to Jupyter Server V2

### DISCUSSION THREAD

**Discussion by HighKeys** [jupyter/enterprise_gateway#1397] [General]
Hi there!

I have been researching the process of upgrading Jupyter Enterprise Gateway (JEG) to Jupyter Server V2, but I couldn't find a summary of the necessary steps and potential challenges. Do you have any resources or information on this topic? I might be interested in contributing to this effort.

Best regards,

**Comment by lresende** [comment]
I believe that would be moving to Provisioners -> https://github.com/jupyter-server/gateway_provisioners ? But what are your concerns or requirements to move to server v2? 

cc @kevin-bates

**Reply by HighKeys** [reply]
While researching Windows Server support, I found that moving to Server v2 is planned, but I can't find the discussion from kevin-bates anymore. Additionally, there are multiple CVEs on the current server version (see [issue #1388](https://github.com/jupyter-server/enterprise_gateway/issues/1388)).

I believe contributing to the Jupyter ecosystem is a great way to get deeper into it. If you think I should focus on another topic, please let me know.

**Reply by kevin-bates** [reply]
Hi @HighKeys - thank you for your interest in contributing to Jupyter and Enterprise Gateway in particular.  (And thank you @lresende for the ping.)

While I am unable to spend time in this area (as much as I'd LOVE to get back to this!), Luciano is correct in that the primary effort of moving JEG (4.0) to JS 2.0 is essentially removing the process-proxy layer and leveraging the gateway provisioners project.  

When I looked into doing this previously (and, I really apologize that I am unable to find that work!), I had run into an additional issue due to the Jupyter Server's changes in the user authorization area which, looking now, may have been introduced with [these changes](https://github.com/jupyter-server/jupyter_server/pull/165).

What should be left of EG (sans process proxies) are essentially the various extensions (like kernelspec caching, some of the "minimal" HA/DR stuff, some user filtering on kernelspecs, etc.) that reside in the layers above process proxies and I would hope that only the parent classes to the Kernel and KernelSpec managers would require updates (along with what is necessary to plumb the authorization stuff).

It might also be good to have EG go ahead and add a dependency on GatewayProvisioners so that the associated entry-points get installed when EG is installed, etc.

I hope that helps.
